### PR TITLE
Test

### PR DESCRIPTION
**Pull request by fortmarek**
### Short description 📝

Continuing work on previous [PR](https://github.com/tuist/tuist/pull/1560)

`test` command should be really useful to ease the interaction with tuist projects.

### Solution 📦

Add a new command that will run all test schemes or you can specify a test scheme by yourself.

The current implementation of running all test schemes (aka running `tuist test` without any additional options) depends on autogenerated schemes - otherwise we could run the same tests multiple times which is not desirable.
I am thinking about ditching `disableAutogeneratedSchemes` option as it's gonna make it harder to optimize `test` (and other commands such as `build` and `run`) going forward. But that is maybe for a separate discussion.

Apart from specifying `scheme` and `configuration`, you can also specify device (`--device "iPhone X"`) and OS version (`--os-verion 14.0`) - the platform is automatically chosen for you. The interface is a little bit different from the original PR, but it would make the implementation of the command a little bit clumsy as you'd need separate option for iOS, all device models, etc. This makes it more future-proof and easier to maintain.

### Implementation 👩‍💻👨‍💻

> Detail in a checklist the steps that you took to implement the PR.

- [x] Finish `TestCommand` and `TestService`
- [x] Automatically choose device to run the tests on (or let the user specify arguments for it)
- [x] Add tests
- [x] Add documentation

### CODE REVIEW

File: features/test.feature [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Feature: Tests projects using Tuist test
+  Scenario: The project is an application with tests (ios_app_with_tests)
+    Given that tuist is available
+    And I have a working directory
+    Then I copy the fixture ios_app_with_tests into the working directory
+    Then tuist tests the project
+    Then tuist tests the scheme AppTests from the project
+    Then tuist tests the scheme App and configuration Debug from the project
```
Comment (pepibumur) [inline features/test.feature:8]: @fortmarek what do you think about having a fixture just for testing purposes, and have a target per platform. That way we can make sure that the testing command works fine for every platform: macOS, iOS, watchOS, tvOS.

**Review by pepibumur — APPROVED** [review]
👏👏👏👏👏👏👏👏👏
Very solid work @fortmarek. It's incredibly well-executed. Congrats!
Besides some minor comments that I left, I think you'll have to update [this array](https://github.com/tuist/tuist/blob/44e5e4506eb5cb7fed1f7da55d39e4caac65a39c/.github/workflows/tuist.yml#L74) to make sure that your acceptance test runs on CI.

File: website/markdown/docs/commands/test.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+**Test a specific scheme**
+
+```bash
+tuist test MyScheme
+```
+
+**Test on a specific device and OS version**
+
+```bash
+tuist test --device "iPhone X" --os-version 14.0
```
[the remark this answers is not in the document]
Comment (fortmarek) [reply website/markdown/docs/commands/test.mdx:36]: yeah, that sounds reasonable 🤞

File: features/test.feature (hunk 1 above)
Comment (fortmarek) [reply features/test.feature:8]: That's a good idea, let's do that 👍

**Pull request merged**